bundle/Controller/Controller.php

```diff
@@ -3,19 +3,13 @@
 namespace Netgen\Bundle\EzPlatformSiteApiBundle\Controller;
 
 use eZ\Bundle\EzPublishCoreBundle\Controller as BaseController;
+use Netgen\EzPlatformSiteApi\Core\Traits\PagerfantaFindTrait;
 use Netgen\EzPlatformSiteApi\Core\Traits\SearchResultExtractorTrait;
 
 abstract class Controller extends BaseController
 {
     use SearchResultExtractorTrait;
-
-    /**
-     * @return \Netgen\EzPlatformSiteApi\API\Site
-     */
-    public function getSite()
```

Comment:
We can't remove that as this class depends on it in `getRootLocation()`, and `PagerfantaFindTrait` should not be required for it to work.

Reply:
As I understood @MarioBlazek, one of the points was to use `PagerfantaFindTrait` inside this controller too, in which case this method is not needed any more since it's available inside the trait itself.

Reply:
That's kind of obvious, but what if we introduce another trait instead:

```php
SiteAwareTrait
{
    public setSite($site);
    protected getSite();
}
```

Then we can use it to implement abstract trait method `PagerfantaFindTrait::getSite()`.

What do you think?

Reply:
+1 for `SiteAwareTrait`, that was my first idea to go this way.

Reply:
`SiteAwareTrait` is fine, but as I understood, you were against it @pspanja ? :)

Reply:
I was? Where?

Reply:
We talked in person :)

Reply:
Oh well :D